File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/helper/internal/TransactionRecord.java
package me.yokeyword.fragmentation.helper.internal;

import android.view.View;

import java.util.ArrayList;

/**
 * @hide Created by YoKey on 16/11/25.
 */
public final class TransactionRecord {
    public String mTag;
    public int mTargetFragmentEnter = Integer.MIN_VALUE;
    public int mCurrentFragmentPopExit = Integer.MIN_VALUE;
    public int mCurrentFragmentPopEnter = Integer.MIN_VALUE;
    public int mTargetFragmentExit = Integer.MIN_VALUE;
    public boolean mDontAddToBackStack = false;
    public ArrayList<SharedElement> mSharedElementList;

    public static class SharedElement {
        public View mSharedElement;
        public String mSharedName;

        public SharedElement(View sharedElement, String sharedName) {
            this.mSharedElement = sharedElement;
            this.mSharedName = sharedName;
        }
    }
}

File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/helper/internal/TransactionRecord.kt
package me.yokeyword.fragmentation.helper.internal

import android.view.View

import java.util.ArrayList

/**
 * @hide Created by YoKey on 16/11/25.
 */
class TransactionRecord {
    var tag: String? = null
    var targetFragmentEnter = Integer.MIN_VALUE
    var currentFragmentPopExit = Integer.MIN_VALUE
    var currentFragmentPopEnter = Integer.MIN_VALUE
    var targetFragmentExit = Integer.MIN_VALUE
    var dontAddToBackStack = false
    var sharedElementList: ArrayList<SharedElement>? = null

    class SharedElement(var sharedElement: View, var sharedName: String)
}
